### DISCUSSION TITLE
spell check in outli headings

### DISCUSSION THREAD

**Discussion by rdiaz02** [minad/jinx#166] [Q&A]
I am using outli: https://github.com/jdtsmith/outli , mainly in prog-mode. To have jinx spell check the headings I have added org-level-1, org-level-2, ..., org-level-8 to `jinx-include-faces` in the faces of `prog-mode`.

```elisp
(setq jinx-include-faces '((prog-mode font-lock-comment-face font-lock-doc-face font-lock-string-face
						;; the next are to have jinx work with outli
						org-level-1 org-level-2 org-level-3 org-level-4
						org-level-5 org-level-6 org-level-7 org-level-8)
                                             (conf-mode font-lock-comment-face font-lock-string-face)
				             (yaml-mode . conf-mode)
				            (yaml-ts-mode . conf-mode)))
```

 This seems to work fine (as the face for the headings is `org-level-1` , `org-level-2`, ...). 

Is this the recommended way of doing it? Or am I missing something? Adding here @jdsmith , in case he regards this as something worth mentioning in outli's README.

**Comment by minad** [comment]
Doesn't outli use outline-1, outline-2 faces and so on instead of the org-level faces?

**Reply by rdiaz02** [reply]
~The weird thing is that the face differs between two machines that are using the exact same emacs-init.el. The one I was using last night shows org-level (and correcly highlights the English spelling error). I just checked, this is a screenshot:~
~But the laptop I am using now has, as you say, `outline-1`, etc.~

Your are right; sorry (I was using, in the maching with `org-outline` , an outli repo that had not been updated since December 2023; I apologize for the confusion).  So we would need to add `outline-1`, ..., to `jinx-include-faces`: this would be the way to do it?

**Reply by minad** [reply]
Not sure, since I haven't tried Outli yet. Maybe @jdtsmith can help?

**Reply by jdtsmith** [reply]
Yep, I recently switched to using the `outline-` faces directly.  On most themes org faces inherit directly from their outline equivalents to help support pre-org bundled Emacsen.  I suppose jinx could look for inheritance.  Face properties on the actual heading text looks like:

```
  face                 (:inherit outline-2 :extend t :overline t)
```

but I'm not sure how jinx can be setup to match that.  I've been using jinx for a bit, but I just let it pick where it wants to check ;).

BTW, unrelated thing: your new `@` orderless annotation-search is fantastic, but makes it harder to select the "add to user dictionary" last jinx item.  Is there a work-around you use?

**Reply by minad** [reply]
Thanks for chiming in!

> I suppose jinx could look for inheritance.

I've already rejected this in some issue (simplicity and performance iirc).

> but I'm not sure how jinx can be setup to match that. I've been using jinx for a bit, but I just let it pick where it wants to check ;).

One has to write a custom Jinx predicate I guess.

> BTW, unrelated thing: your new @ orderless annotation-search is fantastic, but makes it harder to select the "add to user dictionary" last jinx item. Is there a work-around you use?

Thanks, but as usual, you are running and outdated version. ;)

**Reply by rdiaz02** [reply]
> Not sure, since I haven't tried Outli yet. Maybe @jdtsmith can help?

Sorry if I was ambiguous: it does work perfectly (i.e., jinx checks headings) by adding `outline-1` ... to `jinx-include-faces`. And this is the way to do it, and this is not an ugly kludge?  That was my question.

And the second question is (provided the answer to the first is "yes"), should  this be mentioned in outli's README?

**Reply by jdtsmith** [reply]
> Thanks, but as usual, you are running and outdated version. ;)

You'd think by now I'd update all of your and Omar's packages before asking.  Fixed, complete with a new mnemonic — _andnotation_.

**Reply by minad** [reply]
> You'd think by now I'd update all of your and Omar's packages before asking. Fixed, complete with a new mnemonic — andnotation.

Yes, please do that. I don't want to sound too annoyed, but it gets a little bit repetitive.

**Reply by jdtsmith** [reply]
I typically don't update too frequently, because that can require me to hunt down and fix incompatibilities that have been introduced, which I don't often have the time for.  But I can make a list of exceptions for high quality packages ;).

**Reply by minad** [reply]
> I typically don't update too frequently, because that can require me to hunt down and fix incompatibilities that have been introduced, which I don't often have the time for. But I can make a list of exceptions for high quality packages ;).

I understand that, but I also value my time. Just don't install any low quality packages. It is quite easy. Packages which break repeatedly are better kicked out. Furthermore the more packages you install, the more potential security holes will open up. Packages from ELPA at least should be relatively safe since changes are inspected from time to time. However even there we don't have proper commit signature checks in place yet.
